PSEUDO API 1.md

```diff
@@ -0,0 +1,272 @@
+
+		attribute EventHandler onchange;
+	};
+
+A Reality represents a view of the world, be it the real world via sensors or a virtual world that is rendered with WebGL or WebGPU.
+
+Realities can be shared among XRSessions, with multiple scripts rendering into their separate XRLayer.context that are then composited by the UA with the Reality being the rearmost layer.
+
+A script can request an empty Reality from the session in order to create a fully virtual environment.
+
+### Todo
+- Need to expose the stage origin and bounds
+- configuration (e.g. change white balance on camera input, change options on map view)
+
+## XRPointCloud
+
+	interface XRPointCloud {
+		readonly attribute Float32Array points;
+	}
+
+## XRLightEstimate
+
+	interface XRLightEstimate {
+		readonly attribute double ambientIntensity;
+		readonly attribute double ambientColorTemperature;
+	}
+
+## XRAnchor
+
+	interface XRAnchor {
+		readonly attribute DOMString uid;
+		readonly attribute XRCoordinateSystem coordinateSystem;
+		readonly attribute double x;
+		readonly attribute double y;
+		readonly attribute double z;
+	}
+
+## XRPlaneAnchor
+
+	interface XRPlaneAnchor : XRAnchor {
+		readonly attribute Float32Array orientation; // quaternion?
+		readonly attribute double width;
+		readonly attribute double length;
+	}
+
+## XRManifold
+
+	interface XRManifold {
+		TBD
+	}
+
+### Todo
+
+- expose the manifold vertices and edges as well as its extent (FOV only, full sphere, etc)
+
+## XRPresentationFrame
+
+	interface XRPresentationFrame {
+		readonly attribute FrozenArray<XRView> views;
+
+		readonly attribute boolean hasPointCloud;
+		readonly attribute XRPointCloud pointCloud;
+
+		readonly attribute boolean hasManifold;
+		readonly attribute XRManifold manifold;
+
+		readonly attribute boolean hasLightEstimate;
+		readonly attribute XRLightEstimate lightEstimate;
+		
+		long addAnchor(XRAnchor anchor);
+		void removeAnchor(DOMString uid);
+		XRAnchor? getAnchor(DOMString uid);
+		<sequence <XRAnchor>> getAnchors();
+
+		XRDisplayPose? getDisplayPose(XRCoordinateSystem coordinateSystem);
+	};
+
+_The XRPresentationFrame differs from the VRPresentationFrame with the addition of the point cloud, manifold, light estimates, and anchor management._
+
+### Todo
+
+- access camera image buffer aor texture
+
+## XRView
+
+	interface XRView {
+		readonly attribute XREye eye;
+		readonly attribute Float32Array projectionMatrix;
+
+		XRViewport? getViewport(XRLayer layer);
+	};
+
+## XRViewport
+
+	interface XRViewport {
+		readonly attribute long x;
+		readonly attribute long y;
+		readonly attribute long width;
+		readonly attribute long height;
+	};
+
+
+## XRCartographicCoordinates
+
+	interface XRCartographicCoordinates {
+		attribute double latitude;
+		attribute double longitude;
+		attribute double positionAccuracy;
+		attribute double altitude;
+		attribute double altitudeAccuracy;
+		attribute Float32Array orientation; // quaternion from 0,0,0,1 EUS?
```

Comment:
Is a quaternion from EUS the appropriate representation for orientation of cartographic coordinates?